pandas-stubs/core/series.pyi

```diff
@@ -360,7 +360,7 @@ class Series(IndexOpsMixin[S1], NDFrame):
         | tuple[Hashable | slice, ...],
     ) -> Self: ...
     @overload
-    def __getitem__(self, idx: int | _str) -> S1: ...
+    def __getitem__(self, idx: int | _str | Timestamp) -> S1: ...
```

Comment:
Is the underlying issue here that Series is not generic in terms of its index's dtype?

As Series is not generic in its index, I would be inclined to widen it to `idx: Any` - can easily come up with other false positives otherwise.

Reply:
I think `Any` would be too wide.   Maybe we should use `Scalar` ?  Thoughts @twoertwein ?

Reply:
`Scalar` sounds like a good idea - that probably also avoids overlapping overloads.

Reply:
@GarrisonD based on the discussion above, can you replace `int | _str` with `Scalar`, merge with `main` and then we can run the test.